examples/C++/src/demo1.cxx

```diff
@@ -55,8 +56,9 @@ int main()
 
   density_sptr->fill(0);
 
-  /////////////// back project
-  BackProjectorByBinUsingInterpolation back_projector;
+  /////////////// back project, using ray tracing
+  shared_ptr<ProjMatrixByBin> projection_matrix(new  ProjMatrixByBinUsingRayTracing());
+  BackProjectorByBinUsingProjMatrixByBin back_projector(projection_matrix); 
```

Comment:
```suggestion
  auto projection_matrix_sptr = std::make_shared_ptr<ProjMatrixByBinUsingRayTracing>();
  BackProjectorByBinUsingProjMatrixByBin back_projector(projection_matrix_sptr);
```
I think this will work and is maybe somewhat cleaner?